code/modules/antagonists/changeling/changeling.dm

```diff
@@ -260,6 +260,10 @@
  */
 /datum/antagonist/changeling/proc/on_life(datum/source, seconds_per_tick, times_fired)
 	SIGNAL_HANDLER
+	if(isliving(owner))
+		var/mob/living/living_owner = owner
+		if(living_owner.fire_stacks && living_owner.on_fire)
+			return // No chemical regen while on fire.
```

Comment:
Untested code detected 

![image](https://github.com/tgstation/tgstation/assets/51863163/9a5c220a-04e5-4148-8c77-909f07032680)

Reply:
```suggestion
	if(isliving(owner.current))
		var/mob/living/living_owner = owner.current
		if(living_owner.fire_stacks && living_owner.on_fire)
			return // No chemical regen while on fire.
```

Reply:
i forgot to test this section